integritybackend/numbers.py

```diff
@@ -2,18 +2,151 @@
+
+        Args:
+            receiver_address: mint NFT to the receiver's wallet address
+            nft_metadata: ERC-721 NFT metadata in JSON format
+            blockchain_name: mint NFT on the specified blockchain
+
+        Raises:
+            any file I/O errors
+            JSON decoding errors
+
+        Returns:
+            NFT token URI and ID. If registration fails it returns None.
+        """
+        r = requests.post(
+            _STARLINGMINTNFT_URL,
+            headers={"Authorization": f"Bearer {config.NUMBERS_API_KEY}"},
+            data=[
+                ("address", receiver_address),
+                ("json", json.dumps(nft_metadata)),
+                ("blockchain_name", blockchain_name),
+            ],
+        )
+
+        if not r.ok:
+            _logger.error("Numbers starlingmintnft failed: %s %s", r.status_code, r.text)
+            return None
+
+        data = r.json()
+
+        if data.get("response") is None:
+            _logger.warning(
+                "Numbers starlingmintnft response did not have the 'response' field: %s", r.text
+            )
+            return None
+        if data["response"].get("tokenURI") is None:
+            _logger.warning(
+                "Numbers starlingmintnft response did not have the 'tokenURI' field: %s", r.text
+            )
+            return None
+        if data["response"].get("tokenId") is None:
+            _logger.warning(
+                "Numbers starlingmintnft response did not have the 'tokenId' field: %s", r.text
+            )
+            return None
+
+        return {
+            'network': blockchain_name,
+            'contractAddress': '0x4Dc5FD335fbC5614b2A1641c8A9F7e4b7cC80AE2',
+            'tokenUri': data["response"]["tokenURI"],
+            'tokenId': data["response"]["tokenId"],
+        }
+
+    @staticmethod
+    def create_asset_tree(asset_filepath,
+                          asset_creator_filepath,
+                          nft_record_filepath='',
+                          abstract=''):
+        file_util = FileUtil()
+        asset_tree = {}
+        asset_tree['assetCid'] = file_util.digest_cidv1(asset_filepath),
+        asset_tree['assetSha256'] = file_util.digest_sha256(asset_filepath)
+        asset_tree['assetCreator'] = file_util.digest_cidv1(asset_creator_filepath)
+        asset_tree['assetTimestampCreated'] = int(os.path.getctime(asset_filepath))
+        asset_tree['license'] = {
+            'name': 'mit',
+            'document': 'https://opensource.org/licenses/MIT'
+        }
+        asset_tree['nftRecord'] = file_util.digest_cidv1(nft_record_filepath)
+        # TODO: please set integrity CID, sha256sum, Ethereum signature if available
+        #asset_tree['integrityCid'] =
+        #asset_tree['integritySha256'] =
+        #asset_tree['integritySignature'] =
+        asset_tree['abstract'] = abstract
+        asset_tree['encodingFormat'] = magic.Magic(mime=True).from_file(asset_filepath)
```

Comment:
In general we're going to know the asset mime type. It would be better to have this as an argument instead of trying to detect it here.